### PR TITLE
[SPARK-10775][UI]add search keywords in history page ui

### PR DESCRIPTION
**Pull request by lianhuiwang**
add search button in history page ui that we can search some applications of keywords,example: date or time.The ui is as the following photo:
historyserver-ui-search

### CODE REVIEW

**Comment by andrewor14** [comment]
@lianhuiwang would you mind rebasing to master?

**Comment by andrewor14** [comment]
@sarutak @vanzin

**Comment by sarutak** [comment]
I'll inspect the details but I'm not sure whether it's useful to enable to search by app id.

File: core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala [hunk 1]
```diff
@@ -74,19 +80,27 @@ private[history] class HistoryPage(parent: HistoryServer) extends WebUIPage("")
...
+                rangeIndices(actualPage - plusOrMinus until actualPage, 1 < _, requestedIncomplete,
+                  requestedQuery)
               val rightSideIndices =
                 rangeIndices(actualPage + 1 to actualPage + plusOrMinus, _ < pageCount,
-                  requestedIncomplete)
+                  requestedIncomplete, requestedQuery)
 
               <h4>
+                <form action="" method="GET">
+                  Search: <input type="text" name="query" value={requestedQuery} />
```
Comment (sarutak) [inline core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala]: It's difficult for users to know the targets searched by keywords. How about set initial message "Input App Names" or like that to the text field and remove "Search" label.
It's just one idea. Someone might have better idea.

File: core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala [hunk 2]
```diff
@@ -74,19 +80,27 @@ private[history] class HistoryPage(parent: HistoryServer) extends WebUIPage("")
...
+                  <input type="hidden" name="showIncomplete" value={requestedIncomplete.toString} />
+                  <input type="submit" value="Submit" />
```
Comment (sarutak) [inline core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala]: I think "search" is better than "Submit".

**Comment by lianhuiwang** [comment]
@andrewor14 Yes, I will rebase to master.

**Comment by lianhuiwang** [comment]
@andrewor14 Now i think that's OK. Can you look at it? Thanks.

**Comment by lianhuiwang** [comment]
test it please

**Comment by lianhuiwang** [comment]
@sarutak Thanks. I have addressed your comments.Now UI is as following:
history-ui

**Comment by lianhuiwang** [comment]
test this please.

**Comment by lianhuiwang** [comment]
@andrewor14 @sarutak Now all tests have passed.

**Comment by tgravescs** [comment]
so in https://issues.apache.org/jira/browse/SPARK-10873 we are looking at changing to use the jquery datatables.  This would give us a bunch of things including a search box for free.  We are close to having a prototype of the history server working.

**Comment by sarutak** [comment]
@lianhuiwang #10648 has already merged and I think it includes features proposed in this PR. So, do you mind closing this PR?

**Comment by lianhuiwang** [comment]
@sarutak Yes, I will close this PR. Thanks.

**Pull request closed without merging**